zulipterminal/ui_tools/buttons.py

```diff
@@ -249,10 +253,23 @@ def __init__(self, user_id: int, email: str) -> None:
         self.email = email
 
 
+StreamData = TypedDict('StreamData', {
+    'stream_id': Optional[int],
+    'stream_name': Optional[str],
+})
+
+ParsedNarrowLink = TypedDict('ParsedNarrowLink', {
+    'narrow': str,
+    'stream': StreamData,
```

Comment:
The following was confusing `mypy`:
```python3
ParsedNarrowLink = TypedDict('ParsedNarrowLink', {
    'narrow': str,
    'stream_id': Optional[int],
    'stream_name': Optional[str],
...
stream_data = cls._decode_stream_data(fragments[2])
parsed_link = dict(narrow='stream', **stream_data)  # type: ParsedNarrowLink
```

So, to assist `mypy` and make things simpler I introduced a `TypedDict` for the stream data and nested it instead.